h2/src/main/org/h2/command/Parser.java

```diff
@@ -1210,6 +1205,14 @@ private MergeUsing parseMergeUsing(Merge oldCommand, int start) {
         return command;
     }
 
+    private static void appendTableWithSchemaAndAlias(StringBuilder buff, Table table, String alias) {
+        buff.append(quoteIdentifier(table.getSchema().getName()))
```

Comment:
It might be simpler and more consistent to use `table.getSQL()`

Reply:
Yes, I forgot that `getSQL()` is not overridden in `Table`. But result will be the exactly the same.

It's better not to construct SQL in `parseMergeUsing()` at all so this fix should be treated as a temporary solution to resolve issue that you posted and additional related issues that I found at least somehow. So for now there is no reason to change this code. If I find time to reorganize `parseMergeUsing()` this helper method will be removed anyway. If not, it can be simplified later.